src/BondEvaluatorQuartic.h

```diff
@@ -0,0 +1,236 @@
+
+        auto sigma(v["sigma"].cast<Scalar>());
+        auto epsilon(v["epsilon"].cast<Scalar>());
+
+        const Scalar sigma_2 = sigma * sigma;
+        const Scalar sigma_4 = sigma_2 * sigma_2;
+        sigma_6 = sigma_2 * sigma_4;
+        epsilon_x_4 = Scalar(4.0) * epsilon;
+        }
+
+    pybind11::dict asDict()
+        {
+        pybind11::dict v;
+        v["k"] = k;
+        v["r_0"] = r_0;
+        v["b_1"] = b_1;
+        v["b_2"] = b_2;
+        v["U_0"] = U_0;
+        v["sigma"] = pow(sigma_6, 1. / 6.);
+        v["epsilon"] = epsilon_x_4 / Scalar(4.);
+        v["delta"] = delta;
+        return v;
+        }
+#endif
+
+    Scalar k;           //!< k parameter for quartic potential strength
+    Scalar r_0;         //!< energy barrier breaking distance
+    Scalar b_1;         //!< quartic tuning parameter #1
+    Scalar b_2;         //!< quartic tuning parameter #2
+    Scalar U_0;         //!< quartic energy barrier to "breaking"
+    Scalar sigma_6;     //!< sigma raised to the power of 6 for WCA
+    Scalar epsilon_x_4; //!< epsilon * 4 for WCA
+    Scalar delta;       //!< delta parameter bond
+    }
+#if HOOMD_LONGREAL_SIZE == 32
+    __attribute__((aligned(16)));
+#else
+    __attribute__((aligned(32)));
+#endif
+
+//! Class for evaluating the quartic bond potential
+/*!
+ * This bond potential follows the functional form
+ *
+ * \begin{eqnarray*}
+ *      U(r) = & k4 ((r - r_{0}) - b_{1})((r - r_{0}) - b_{2})(r - r_{0})^2
+ *                 + U_{0} + U_{\rm WCA}(r)      & r < r_0\\
+ *            = & U_0                            & r \ge r_0
+ * \end{eqnarray*}
+ *
+ * which has a WCA repulsive force at close range, and a quartic energy well. On the far side of
+ * this energy well, there is an energy barrier, to a plateau energy, U_0. It reaches this energy
+ * at r_0. Beyond r_0, the energy is constant, and there is no force from the bond potential.
+ *
+ * Since this potential has a finite range, beyond which no force is applied, this potential is
+ * useful for modeling bonds with scissile functionality. r_0 defines the bond length at which
+ * the bond can be considered broken, and U_0 defines the height of the energy barrier to breaking.
+ *
+ * The parameters b_1 and b_2 tune the location of the energy minimum, as well as the well width.
+ * They do not cleanly translate to meaningful locations on the potential energy surface, but serve
+ * as helpful tuning parameters to mimic other bond topologies. For example, making a previously
+ * FENE or harmonic bond breakable. Using linear regression or some other fitting algorithm may do
+ * this.
+ *
+ * The delta parameter shifts the whole potential with respect to r. The default value is 0.
+ * Positive values shift to higher r values.
+ */
+class BondEvaluatorQuartic : public BondEvaluator
+    {
+    public:
+    typedef BondParametersQuartic param_type;
+
+    DEVICE BondEvaluatorQuartic(Scalar _rsq, const param_type& _params)
```

Comment:
This is a lot of parameters. While equivalent, I would prefer if you could assign them inside the constructor, like for example here: https://github.com/mphowardlab/azplugins/blob/0c520e05afe59ef07ab3405fccaff4d90524fc45/src/DPDPairEvaluatorGeneralWeight.h#L109